src/renderer.ts

```diff
@@ -152,6 +158,504 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+      boxdata.median = scale(boxdata.median);
+      boxdata.q1 = scale(boxdata.q1);
+      boxdata.q3 = scale(boxdata.q3);
+      const minPos = boxdata.min, maxPos = boxdata.max, medianPos = boxdata.median, q3Pos = boxdata.q3, q1Pos = boxdata.q1;
+      ctx.fillStyle = '#e0e0e0';
+      ctx.strokeStyle = 'black';
+      ctx.beginPath();
+      ctx.rect((q1Pos), topPadding, ((q3Pos) - (q1Pos)), (context.rowHeight(i) - (topPadding * 2)));
+      ctx.fill();
+      ctx.stroke();
+
+      //Line
+      const bottomPos = (context.rowHeight(i) - topPadding);
+      const middlePos = (context.rowHeight(i) - topPadding) / 2;
+
+      ctx.strokeStyle = 'black';
+      ctx.fillStyle = '#e0e0e0';
+      ctx.beginPath();
+      ctx.moveTo(minPos, middlePos);
+      ctx.lineTo((q1Pos), middlePos);
+      ctx.moveTo(minPos, topPadding);
+      ctx.lineTo(minPos, bottomPos);
+      ctx.moveTo(medianPos, topPadding);
+      ctx.lineTo(medianPos, bottomPos);
+      ctx.moveTo((q3Pos), middlePos);
+      ctx.lineTo(maxPos, middlePos);
+      ctx.moveTo(maxPos, topPadding);
+      ctx.lineTo(maxPos, bottomPos);
+      ctx.stroke();
+      ctx.fill();
+
+
+      if (currentSortColumn === col) {
+
+        ctx.strokeStyle = 'red';
+        ctx.fillStyle = '#ff0700';
+        ctx.beginPath();
+        ctx.moveTo(boxdata[userSort], topPadding);
+        ctx.lineTo(boxdata[userSort], bottomPos);
+        ctx.stroke();
+        ctx.fill();
+      } else {
+        return;
+      }
+
+
+    };
+  }
+
+}
+
+function setPathCalculate(setData, cellDimension) {
+
+  const catindexes = [];
+  setData.forEach((d, i) => (d === SetColumn.IN_GROUP) ? catindexes.push(i) : -1);
+
+
+  const leftX = ((d3.min(catindexes) * cellDimension) + (cellDimension / 2));
+  const rightX = ((d3.max(catindexes) * cellDimension) + (cellDimension / 2));
+
+  const pathdata = {left: leftX, right: rightX};
+
+  return pathdata;
+
+}
+
+
+class SetCellRenderer implements ICellRendererFactory {
+
+  createSVG(col: SetColumn, context: IDOMRenderContext): ISVGCellRenderer {
+    const cellDimension = col.cellDimension();
+    const binaryValue = col.getConstantValue();
+    return {
+      template: `<g class="upsetcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const circle = d3.select(n).selectAll('circle').data(col.getValue(d.v, d.dataIndex));
+        circle.enter().append('circle');
+        circle
+          .attr({
+            cy: (d: any, j) => (context.rowHeight(j) / 2),
+            cx: (d: any, j) => (j * cellDimension) + (cellDimension / 2),
+            r: (cellDimension / 4),
+            class: (d) => (d === binaryValue) ? 'setcircle' : 'setcircleOpacity'
+          });
+        circle.exit().remove();
+        const path = d3.select(n).selectAll('path').data(<any>[col.getValue(d.v, d.dataIndex)]);
+        const countCategory = col.getValue(d.v, d.dataIndex).filter((x) => x === binaryValue).length;
+        if (countCategory > 1) {
```

Comment:
what happens if in the first update the `countCategory` is > 1 and in a later run with different data it isn't?